File: src/controllers/appointment.controller.js
const appointmentService = require('../services/appointment.service');

async function create(req, res, next) {
  try {
    const appointment = await appointmentService.create({
      ...req.body,
      cliente_id: req.user.id
    });

    res.status(201).json(appointment);
  } catch (error) {
    next(error);
  }
}

async function list(req, res, next) {
  try {
    const tipo = req.query.tipo || 'ativos';
    const appointments = await appointmentService.listAll(req.user, tipo);
    res.json(appointments);
  } catch (error) {
    next(error);
  }
}

async function updateStatus(req, res, next) {
  try {
    const updated = await appointmentService.updateStatus(
      req.params.id,
      req.body.status
    );

    res.json(updated);
  } catch (error) {
    next(error);
  }
}

async function remove(req, res, next) {
  try {
    const deleted = await appointmentService.remove(req.params.id);
    res.json(deleted);
  } catch (error) {
    next(error);
  }
}

module.exports = {
  create,
  list,
  updateStatus,
  remove
};

File: Downloads/BarbeariaTerminal-migracao/BarbeariaTerminal-migracao/Backend/barberproapi/src/main/java/com/barberpro/barberproapi/controller/AppointmentController.java
package com.barberpro.barberproapi.controller;

import com.barberpro.barberproapi.domain.User;
import com.barberpro.barberproapi.dto.AppointmentRequest;
import com.barberpro.barberproapi.dto.AppointmentResponse;
import com.barberpro.barberproapi.dto.AppointmentStatusRequest;
import com.barberpro.barberproapi.service.AppointmentService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.security.core.Authentication;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/appointments")
@RequiredArgsConstructor
public class AppointmentController {

    private final AppointmentService appointmentService;

    @GetMapping
    public List<AppointmentResponse> list(
            Authentication authentication,
            @RequestParam(defaultValue = "ativos") String tipo
    ) {
        User user = (User) authentication.getPrincipal();
        return appointmentService.listAll(user, tipo);
    }

    @PostMapping
    @PreAuthorize("hasRole('CLIENTE')")
    public ResponseEntity<AppointmentResponse> create(
            @Valid @RequestBody AppointmentRequest request,
            Authentication authentication
    ) {
        User user = (User) authentication.getPrincipal();
        return ResponseEntity.status(HttpStatus.CREATED).body(appointmentService.create(request, user));
    }

    @PatchMapping("/{id}/status")
    @PreAuthorize("hasAnyRole('ADMIN', 'BARBEIRO')")
    public AppointmentResponse updateStatus(
            @PathVariable Long id,
            @Valid @RequestBody AppointmentStatusRequest request
    ) {
        return appointmentService.updateStatus(id, request.status());
    }

    @DeleteMapping("/{id}")
    public AppointmentResponse remove(@PathVariable Long id, Authentication authentication) {
        User user = (User) authentication.getPrincipal();
        return appointmentService.remove(id, user);
    }

    @GetMapping("/disponibilidade")
    public List<String> disponibilidade(
            @RequestParam Long barbeiroId,
            @RequestParam Long serviceId,
            @RequestParam String data
    ) {
        return appointmentService.horariosDisponiveis(
                barbeiroId,
                serviceId,
                java.time.LocalDate.parse(data)
        );
    }
}